### PR TITLE
Feature/add multiple package scanning

### PR DESCRIPTION
**Pull request by ruben-perez**
Allow multiple values (comma-separated)  for the "com.scmspain.karyon.rest.property.packages" property. With that enabled, we could have a project specific package for its own controllers, and also the health check controller provided by the [karyon2-healthcheck](https://github.com/scm-spain/karyon2-healthcheck) project.

### CODE REVIEW

**Comment by victuxbb** [comment]
Hi Ruben! I think maybe will be better to use a prefix in properties files to be able to create multiple paths to scan instead of a comma separated values, let me put an example:

```
com.scmspain.karyon.rest.property.packages.controllers=com.scmspain.lalla.ninja.controllers
com.scmspain.karyon.rest.property.packages.healthchecks=com.scmspain.yoyo.healthchecks

```
And then in the code, get all properties with the beginning match:

```java
ConfigurationManager.getConfigInstance().subset("com.scmspain.karyon.rest.property.packages")
```

What do you think? :)

**Comment by ruben-perez** [comment]
I didn't think about the subset capability, so thank you for that :)

However, I don't think this is a good use case for that. I'll try to explain myself with an example: Let's suppose we have a properties file with this content:

```
xxx.yyy.zzz.asd=value1
xxx.yyy.zzz.qwerty=value2
xxx.yyy.zzz.zxc=value3
aaa.sss.ddd=value4
```

If I had that, I would expect those properties to be read as: 
* "xxx.yyy.zzz" is the prefix that marks who will load and handle those properties. That gives us context about the type/meaning of the property, and who will handle them.
* "asd", "qwerty" and "zxc" would be the meaningful keys for the handling application. Those should be choosen from a pool of accepted (even needed) keys for that application, not randomly chosen by the "client" application where those values are written. 
* "value1", "value2", "value3" are the exact values for the above keys, and is important the key-value relation, they are not interchangeable. 

In the case of this PR, I don't think we have that scenario. In our case, we need a list of packages to be scanned, with no distinction or difference in the way we handle them. The way I would read these properties following the above explanation would be:

```
com.scmspain.karyon.rest.property.packages.controllers=com.scmspain.lalla.ninja.controllers
com.scmspain.karyon.rest.property.packages.healthchecks=com.scmspain.yoyo.healthchecks
```

* Someone will load/handle the properties with the *com.scmspain.karyon.rest.property.packages* preffix, as those make sense as a group for someone.
* Then, we would have the meaningful key-value properties:
```
controllers=com.scmspain.lalla.ninja.controllers
healthchecks=com.scmspain.yoyo.healthchecks
```
But those are not actual properties of our application, as far as I see them. Those are not keys for our properties, the *controllers* and *healthchecks* keys are not relevant or even valid. 

As far as I see it, a better subset use case example would be:

```
com.app.port=my port
com.app.host=my host
com.app.path=my path
```

Because *port*, *host* and *path* would be relevant properties for my application.

I hope I've been able to explain my point of view well enough... What do you think?

Anyway, this subset suggestion made me think, and I will use it when I see it as a good fit for sure :)

**Comment by txun** [comment]
In my opinion, it is **ok** having multiple package scanning support. 
People must be able to put their controllers in multiple packages and karyon-rest-router find those controllers.
This is not exclusive with having subset config feature (also it is a good feature), although the **ruben-perez** use case really don't need this. In this use case the healthcheck concept is not really important, instead is the **routing** of the healthcheck, which in this context it's a simple controller with some endpoint declared.
<blockquote>
In conclusion, having multiple package scanning support in karyon-rest-router, makes it more flexible .
</blockquote>

**Comment by txun** [comment]
:+1:

**Comment by borjavh** [comment]
I'm in agree with the Justo comment. For me having multiple package scanning support is a good feature and give value by itself. 

We can discuss later if is needed to use a specific property for the healthCheck controller or for another specific controllers.

:+1:

**Comment by fiunchinho** [comment]
:+1:

**Comment by victuxbb** [comment]
Ok sorry, I didn't see that you are using the native method `getStringArray`from ConfigurationManager for parsing multiples values.
For me it's :+1:
